### PR TITLE
Series, models and api

### PR DESCRIPTION
**Pull request by SibiAkkash**

### CODE REVIEW

File: server/series/models.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+class SeriesRosterInvitation(ExportModelOperationsMixin("series_roster_invitation"), models.Model):  # type: ignore[misc]
+    class Status(models.TextChoices):
+        PENDING = "Pending", _("Pending")
+        ACCEPTED = "Accepted", _("Accepted")
+        DECLINED = "Declined", _("Declined")
+        EXPIRED = "Expired", _("Expired")
+
+    series = models.ForeignKey(Series, on_delete=models.CASCADE)
+    from_player = models.ForeignKey(
```
Comment (Joe2k) [inline server/series/models.py:60]: should be from_user.. I can just be normal user with no player info and still add you in my team

File: server/series/models.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        Player, on_delete=models.CASCADE, related_name="series_invitations_sent"
```
Comment (Joe2k) [inline server/series/models.py:61]: on delete we should be careful.. better would be do nothing?

File: server/series/models.py (hunk 1 above)
Comment (SibiAkkash) [reply server/series/models.py:60]: ah yes, should be from_user

File: server/series/models.py (hunk 2 above)
Comment (SibiAkkash) [reply server/series/models.py:61]: hmm. we could leave it be i guess, we wouldn't know who sent the invitation, if thats fine

Comment (Joe2k) [reply server/series/models.py:61]: ok this is just invitation.. so cascade is fine.. For proper registration we should be careful

Comment (SibiAkkash) [reply server/series/models.py:61]: yeah, thats why i left. will not do this for registration

File: server/tournament/models.py [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+    team_registration_end_date = models.DateField()
+    player_registration_start_date = models.DateField()
+    player_registration_end_date = models.DateField()
+    ultimate_central_id = models.PositiveIntegerField(unique=True, null=True, blank=True)
+    ultimate_central_slug = models.SlugField(max_length=200, default="unknown")
+    location = models.CharField(max_length=255, default="unknown")
+    type = models.CharField(max_length=3, choices=Type.choices, default=Type.MIXED)
+    slug = models.SlugField(null=True, blank=True, db_index=True)
+    max_num_teams = models.PositiveSmallIntegerField(blank=True, null=True)
+    series = models.OneToOneField(Series, on_delete=models.SET_NULL, blank=True, null=True)
```
Comment (Joe2k) [inline server/tournament/models.py:33]: Why onetoone? should be foreign key?

File: server/tournament/models.py [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+    qualifier_event_id = models.OneToOneField(
```
Comment (Joe2k) [inline server/tournament/models.py:34]: this is not needed for now I think.. And this ideally should be list as South Regionals will have TN, KA, KL Sectionals as qualifier event..

So yeah I dont think itll be much useful now anyway?

File: server/tournament/models.py [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+    spirit_ranking = models.JSONField(default=list)
+
+    use_uc_registrations = models.BooleanField(default=False)
+
+
+@receiver(m2m_changed, sender=Tournament.teams.through)
+def update_seeding_on_teams_change(
+    sender: Any, instance: Tournament, action: str, **kwargs: Any
+) -> None:
+    if action == "post_add":
```
Comment (Joe2k) [inline server/tournament/models.py:83]: existing bug, fix if possible: have to add "post_remove" cause when we remove team registration we need to remove from seedings as well

File: server/tournament/models.py [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+class Registration(ExportModelOperationsMixin("registration"), models.Model):  # type: ignore[misc]
+    class Role(models.TextChoices):
+        DEFAULT = "DFLT", _("Default")
+        CAPTAIN = "CAP", _("Captain")
+        SPIRIT_CAPTAIN = "SCAP", _("Spirit Captain")
+        COACH = "COACH", _("Coach")
+        ASSISTANT_COACH = "ACOACH", _("Assistant Coach")
+        MANAGER = "MNGR", _("Manager")
+
+    series_registration = models.OneToOneField(
```
Comment (Joe2k) [inline server/tournament/models.py:122]: why one to one? Should be foreign?

File: server/tournament/models.py [hunk 7]
```diff
@@ -0,0 +1,307 @@
...
+
+class EventRosterInvitation(ExportModelOperationsMixin("event_roster_invitation"), models.Model):  # type: ignore[misc]
+    class Status(models.TextChoices):
+        PENDING = "Pending", _("Pending")
+        ACCEPTED = "Accepted", _("Accepted")
+        DECLINED = "Declined", _("Declined")
+        EXPIRED = "Expired", _("Expired")
+
+    event = models.ForeignKey(Event, on_delete=models.CASCADE)
+    from_player = models.ForeignKey(
```
Comment (Joe2k) [inline server/tournament/models.py:102]: same from_user

File: server/tournament/models.py (hunk 5 above)
Comment (Joe2k) [reply server/tournament/models.py:83]: https://docs.djangoproject.com/en/5.0/ref/signals/#m2m-changed

File: server/tournament/models.py [hunk 8]
```diff
@@ -0,0 +1,314 @@
...
+        seeding = {}
+
+        for i, team in enumerate(instance.teams.all(), start=1):
+            seeding[i] = team.id
+
+        instance.initial_seeding = seeding
+        instance.current_seeding = seeding
+        instance.save()
+
+    if action == "post_remove":
```
Comment (Joe2k) [inline server/tournament/models.py:90]: minor comment: Why added this seperately? Can just combine in the top to have if action is post_add or post_remove do the same thing?

**Review by Joe2k — APPROVED** [review]

**Pull request merged**